lib/test/service.go

```diff
@@ -122,3 +122,15 @@ func ValidateServiceResources(r *KnRunResultCollector, serviceName string, reque
 	serviceLimitsResourceList := service.Spec.Template.Spec.Containers[0].Resources.Limits
 	assert.DeepEqual(r.T(), serviceLimitsResourceList, llist)
 }
+
+//GetServiceFromKNServiceDescribe runs the kn service describe command
+//decodes it into a ksvc and returns it.
+func GetServiceFromKNServiceDescribe(r *KnRunResultCollector, serviceName string) servingv1.Service {
+	out := r.KnTest().Kn().Run("service", "describe", serviceName, "-ojson")
+	data := json.NewDecoder(strings.NewReader(out.Stdout))
+	data.UseNumber()
+	var service servingv1.Service
+	err := data.Decode(&service)
+	assert.NilError(r.T(), err)
+	return service
```

Comment:
we can return here the pointer to service object instead

Reply:
(can be fixed in a subsequent PR as well)

Reply:
Nice catch. Let's get this in and address in follow-up PR then.